trie/stacktrie.go

```diff
@@ -107,7 +110,9 @@ func (t *StackTrie) Update(key, value []byte) error {
 	} else {
 		t.last = append(t.last[:0], k...) // reuse key slice
 	}
-	t.insert(t.root, k, value, nil)
```

Comment:
RIght, so, first of all: if insert fails, we're in a pretty shitty state. The internals may be broken, in so far as the structure may have been modified before the error occurred. 

The `insert` can only fail for two reasons: 

- `return errors.New("trying to insert into existing key")` -- this cannot realistically happen if we track `first` properly. 
- `return errors.New("trying to insert into hash")` -- this can also not happen, unless the key ordering is wrong, and we go back. 

The reason I wanted these changes, is to be able to use the stacktrie with _untrusted_ data, via range verification. However, thinking about it now, I believe that this might no longer be needed, if the public `Update` function always prevents them.

Reply:
I do want the errors returned from the `Update` function though, I'll reopen this PR with only that, but no changes in `insert`